core/src/main/scala/com/pingcap/tispark/write/TiBatchWrite.scala

```diff
@@ -146,15 +149,39 @@ class TiBatchWrite(
 
     // init tiBatchWriteTables
     tiBatchWriteTables = {
-      dataToWrite.map {
+      dataToWrite.flatMap {
         case (dbTable, df) =>
-          new TiBatchWriteTable(
-            df,
-            tiContext,
-            options.setDBTable(dbTable),
-            tiConf,
-            tiDBJDBCClient,
-            isTiDBV4)
+          val tableOptions = options.setDBTable(dbTable)
+          val tiTableRef = tableOptions.getTiTableRef(tiConf)
+          val tiTableInfo = tiContext.tiSession.getCatalog.getTable(
+            tableOptions.getTiTableRef(tiConf).databaseName,
+            tableOptions.getTiTableRef(tiConf).tableName)
+
+          if (tiTableInfo == null) {
+            throw new NoSuchTableException(tiTableRef.databaseName, tiTableRef.tableName)
+          }
```

Comment:
Table check has been done in `TiDBWriter.write`? Is here a duplicate?

Reply:
@shiyuhang0 Will this method has another entry rather than TiDBWriter.write()?

Reply:
multi-table batch write may entry this method without check
(https://github.com/pingcap/tispark/blob/master/docs/datasource_api_userguide.md#writing-to-multi-tables)

Reply:
@Daemonxiao So this is necessary